### PR TITLE
fix: clarify `HashOp::Keccak` variant to refer to Keccak-256

### PR DESCRIPTION
**Pull request by Farhad-Shabani**
Closes: #211

### Description
Considering that no use cases have been reported for the current SHA-3 Rust implementation of the `HashOp::Keccak`, and with the goal of bolstering support for meaningful use cases, I have opted for the inclusion of Keccak-256 as the most suitable variant (Ethereum uses this particular hash function).

### Note
For the JS part, I couldn't properly generate code through existing commands, so it made me to adjust things there manually. That would be great if @webmaster128 could look into it.

### CODE REVIEW

**Comment by webmaster128** [comment]
> For the JS part, I couldn't properly generate code through existing commands, so it made me to adjust things there manually. That would be great if @webmaster128 could look into it.

Diff looks sane as far as I can tell. I'd not invest time in the current proto codegen for JS anymore. We should replace that with Telescope code generation.

File: rust/src/host_functions.rs [hunk 1]
```diff
@@ -48,9 +48,9 @@ pub mod host_functions_impl {
             buf
         }
 
-        fn sha3_512(message: &[u8]) -> [u8; 64] {
```
Comment (crodriguezvega) [inline rust/src/host_functions.rs:51]: This is being removed because it's not used?

Comment (Farhad-Shabani) [reply rust/src/host_functions.rs:51]: That’s right. As far as I could check in projects around `ibc-rs`, they are not using it, and there isn't any `HashOp` variant already for it.

**Comment by hdevalence** [comment]
> Considering that no use cases have been reported for the current SHA-3 Rust implementation of the HashOp::Keccak, and with the goal of bolstering support for meaningful use cases, I have opted for the inclusion of Keccak-256 as the most suitable variant (Ethereum uses this particular hash function).

Wait, but SHA-3 is the standard one. It differs from Keccak-256 only in some minor constants. Wouldn't it be better to have support for both standard SHA-3 and the Ethereum-specific variant, considering that there's effectively no extra code required?

**Comment by Farhad-Shabani** [comment]
> Wait, but SHA-3 is the standard one. It differs from Keccak-256 only in some minor constants. Wouldn't it be better to have support for both standard SHA-3 and the Ethereum-specific variant, considering that there's effectively no extra code required?

I'm cool with adding a standard option. However, just to be clear, the SHA-3 implementation we've had so far is SHA3-512 (64 bytes). Afaik, the one that's more in line with Keccak256 is SHA3-256.
Honestly, I haven't come across any major projects using SHA3-512. Maybe we could focus on nailing down the Keccak variant first here and then tackle supporting SHA3 as a separate effort after some evaluation and seeing which algorithm covers the most relevant use cases. What do you think?

**Review by crodriguezvega — APPROVED** [review]
Thank you, @Farhad-Shabani. Sorry it took a while to review. 

The changes look good to me, I just added a [test](https://github.com/cosmos/ics23/pull/217/commits/3ad31165332d2360c3f6e4b2115d4adb53611ac5).

**Comment by Farhad-Shabani** [comment]
> Thank you, @Farhad-Shabani. Sorry it took a while to review.

Thank you so much @crodriguezvega for reviewing this :pray:

**Review by romac — APPROVED** [review]
LGTM

**Pull request merged**